chore(docs): update info on router versions (#23075)
expo/expo: docs/pages/router/reference/async-routes.mdx

import { Terminal } from '~/ui/components/Snippet';
import Video from '~/components/plugins/Video';

> > First available in Expo SDK 49 and Expo Router v2.

Reviewer: ```suggestion
> Available from Expo SDK 49 and Expo Router v2.
```

Reviewer: Just a suggestion for the longevity of these docs.
